tooling/lsp/src/notifications/mod.rs

```diff
@@ -255,13 +265,62 @@ fn file_diagnostic_to_diagnostic(file_diagnostic: FileDiagnostic, files: &FileMa
         tags.push(DiagnosticTag::DEPRECATED);
     }
 
-    Diagnostic {
+    let mut related_information = Vec::new();
+
+    for secondary in diagnostic.secondaries {
+        let secondary_file = secondary.file.unwrap_or(file_id);
+        let Some(path) = fm.path(secondary_file) else {
+            continue;
+        };
+        let Ok(uri) = Url::from_file_path(path) else {
+            continue;
+        };
+        let Some(range) = byte_span_to_range(files, file_id, secondary.span.into()) else {
+            continue;
+        };
```

Comment:
nit: could these three `let _ = _ else { continue }` be combined into something like:
```
let Some((uri, range)) = {
  let path = fm.path(..)?;
  let uri = ...ok()?;
  let range = ..?;
  Some((uri, range))
} else {
  continue;
}
```
?

Reply:
I'm not sure... if you use `?` inside a loop, does it continue the loop or return from the function? In this case I need to continue. I could extract the loop body into a separate function and use `?` there 🤔 I'll try that.

Reply:
`?` would be invalid here since it'd try to return from the outer function. You could maybe wrap it in a closure that is immediately invoked though.

Reply:
I introduced some helper functions to be able to use `?`. I think the code ended up being clearer, thanks!